server/webhooks/registration-parser.js

```diff
@@ -0,0 +1,84 @@
+"use strict";
+import 'babel-register';
+import _ from 'lodash'
+
+class cashnetJSON {
+    constructor(cnj) {
+        let { itemcnt } = cnj;
+        let refs = this.refs(cnj);
+        this.batchNumber = cnj['batchno'];
+        this.primaryContact = {
+            "email": cnj['MA3-PUEMAIL'],
+            "name": cnj['MA3-PUNAME']
+        };
+        this.participants = this.getParticipants(refs);
+        this.tshirts = this.getTshirts(refs);
+        this.transaction = cnj['tx'];
+        this.totalAmount = this.getTotalAmount(refs);
+    }
+    refs(cnj) {
+        let { itemcnt } = cnj;
+        let refArray = [];
+        _.forEach(_.range(1, _.toInteger(itemcnt) + 1), function (index) {
```

Comment:
*  A note on iterating.  Always prefer `map` or `transform` if you can before `forEach` they are faster.
* `Array.push` is slow.
* This loop is the same as:
```javascript
for(let i = 0; i < itemcnt; i++)
// even cleaner with lodash
_.times(itemcnt, (i) => { } )
```
* Faster to declare and instantiate your Array with a size at the same time.